segments/output/mongodb/mongo.go

```diff
@@ -0,0 +1,310 @@
+	collectionName string // default ringbuffer
+	Fields         string // optional comma-separated list of fields to export, default is "", meaning all fields
+	BatchSize      int    // optional how many flows to hold in memory between INSERTs, default is 1000
+}
+
+// Every Segment must implement a New method, even if there isn't any config
+// it is interested in.
+func (segment Mongodb) New(configx map[string]string) segments.Segment {
+	newsegment := &Mongodb{}
+
+	newsegment, err := fillSegmentWithConfig(newsegment, configx)
+	if err != nil {
+		log.Printf("[error] '%s'", err.Error())
+		return nil
+	}
+
+	ctx := context.Background()
+
+	//Test if db connection works
+	client, err := mongo.Connect(ctx, options.Client().ApplyURI(newsegment.mongodbUri))
+	if err == nil {
+		//test if the connection was acutally sucessful
+		err = client.Ping(ctx, options.Client().ReadPreference)
+	}
+	if err != nil {
+		log.Printf("[error] mongoDB: Could not open DB connection due to '%s", err.Error())
+		return nil
+	}
+	db := client.Database(newsegment.databaseName)
+
+	// collection in the mongdo should be capped to limit the used disk space
+	convertToCappedCollection(db, newsegment)
+	return newsegment
+}
+
+func (segment *Mongodb) Run(wg *sync.WaitGroup) {
+	ctx := context.Background()
+	defer func() {
+		close(segment.Out)
+		wg.Done()
+	}()
+
+	client, err := mongo.Connect(ctx, options.Client().ApplyURI(segment.mongodbUri))
+	if err != nil {
+		log.Panic(err) // this has already been checked in New
+	}
+	db := client.Database(segment.databaseName)
+	segment.dbCollection = db.Collection(segment.collectionName)
+
+	defer client.Disconnect(ctx)
+
+	var unsaved []*pb.EnrichedFlow
+
+	for msg := range segment.In {
+		unsaved = append(unsaved, msg)
+		if len(unsaved) >= segment.BatchSize {
+			err := segment.bulkInsert(unsaved, ctx)
+			if err != nil {
+				log.Printf("[error] %s", err)
+			}
+			unsaved = []*pb.EnrichedFlow{}
+		}
+		segment.Out <- msg
+	}
+	segment.bulkInsert(unsaved, ctx)
+}
+
+func fillSegmentWithConfig(newsegment *Mongodb, config map[string]string) (*Mongodb, error) {
+	if config == nil {
+		return newsegment, errors.New("missing configuration for segment mongodb")
+	}
+
+	if config["mongodb_uri"] == "" {
+		return newsegment, errors.New("mongoDB: mongodb_uri not defined")
+	}
+	newsegment.mongodbUri = config["mongodb_uri"]
+
+	if config["database"] == "" {
+		log.Println("[INFO] mongoDB: no database defined - using default value (flowdata)")
+		config["database"] = "flowdata"
+	}
+	newsegment.databaseName = config["database"]
+
+	if config["collection"] == "" {
+		log.Println("[INFO] mongoDB: no collection defined - using default value (ringbuffer)")
+		config["collection"] = "ringbuffer"
+	}
+	newsegment.collectionName = config["collection"]
+
+	var ringbufferSize int64 = 10737418240
+	if config["max_disk_usage"] == "" {
+		log.Println("[INFO] mongoDB: no ring buffer size defined - using default value (10GB)")
```

Comment:
It might be time to rethink config parsing or even add some config parsing library in the future. I did not know enough about Go back in the day, but this sure is a lot of manual work.